src/shared/user-record.c

```diff
@@ -48,11 +50,29 @@ static int read_login_defs(login_defs *ret_defs, const char *path) {
                 if (r == 0)
                         break;
 
-                if ((t = startswith(line, "SYS_UID_MAX")) && strchr(WHITESPACE, t[0])) {
+                if ((t = startswith(line, "SYS_UID_MIN")) && strchr(WHITESPACE, t[0])) {
+                        t += strspn(t, WHITESPACE);
+                        r = parse_uid(t, &sys_uid_min);
+                        if (r < 0)
+                                log_debug_errno(r, "%s: failed to parse SYS_UID_MIN, ignoring: %m", path);
+                        if (sys_uid_min == 0) {
```

Comment:
hmm, i don#t think we should correct 0 → 101 here. This is about allocation, and I think in most setups allocating from the whole < 1000 range is totally ok. i.e. it might make sense to bump 0 to 1 here, since we know for sure that uid 0 is taken by some user and hence not useful as space to allocate from, but conceptually from the distance I think it totally makes sense to say 0…999 is the system uid range that can be allocated from, even though we know that 0 is definitely always already taken...

Hence, maybe upgrade sys_uid_min=0 → sys_uid_min=1, as kind of an optimization for the allocation algorithms, but beyond that i wouldn't do anything here.